### PR TITLE
Merge builder-http-gateway into builder-api

### PR DESCRIPTION
**Pull request by chefsalim**
In the continuing saga of the Builder API cleanup, this change merges the builder-http-gateway crate into builder API.  This allows us to consolidate all usage of iron into builder-api, eliminate un-needed abstractions, and set the stage for further refactoring. Yet more to come..

Signed-off-by: Salim Alam <salam@chef.io>

tenor-193838357

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/Cargo.toml [hunk 1]
```diff
@@ -35,7 +35,7 @@ serde_json = "*"
 staticfile = "*"
 toml = { version = "*", default-features = false }
 typemap = "*"
-unicase = "*"
+unicase = "1"
```
Comment (raskchanky) [inline components/builder-api/Cargo.toml]: Is this intentional?

Comment (chefsalim) [reply components/builder-api/Cargo.toml]: Yeah there is a compatibility issue that is requiring a "1" for now... will fix/update over the course of the upcoming refactors

**Pull request merged**